include/bluetooth/att.h

```diff
@@ -59,12 +59,52 @@ extern "C" {
 #if defined(CONFIG_BT_TESTING)
 
 int bt_eatt_disconnect_one(struct bt_conn *conn);
-int bt_eatt_connect(struct bt_conn *conn, uint8_t num_channels);
 size_t bt_eatt_count(struct bt_conn *conn);
 
 #endif /* CONFIG_BT_TESTING */
+
+/** @brief Connect Enhanced ATT channels
+ *
+ * Sends a series of Credit Based Connection Requests to connect @p num_channels
+ * Enhanced ATT channels. The peer may have limited resources and fewer channels
+ * may be created.
+ *
+ * @param conn The connection to send the request on
+ * @param num_channels The number of Enhanced ATT beares to request.
+ * Must be in the range 1 - @kconfig[CONFIG_BT_EATT_MAX], inclusive.
+ *
+ * @return 0 in case of success or negative value in case of error.
+ * @retval -EINVAL if @p num_channels is not in the allowed range.
+ * @retval -EINVAL if @p conn is NULL.
+ * @retval -ENOMEM if less than @p num_channels are allocated.
+ * @retval 0 in case of success
+ */
+int bt_eatt_connect(struct bt_conn *conn, uint8_t num_channels);
+
 #endif /* CONFIG_BT_EATT */
 
+/* ATT channel information */
+struct bt_att_chan_info {
+	/* The connection the ATT channel belongs to */
+	struct bt_conn *conn;
+};
```

Comment:
What kind of things do you foresee us extending this struct with?

Why is it called `bt_att_chan_info` instead of `bt_eatt_chan_info`?

Reply:
Initially the callback was also called for the fixed ATT channel, hence the name.

There were more fields in the struct, the MTU, CID and MPS, but they were removed. See https://github.com/zephyrproject-rtos/zephyr/pull/43112#r813145144

Reply:
Bump @jhedberg